src/probe_identity.rs

```diff
@@ -0,0 +1,186 @@
+use color_eyre::eyre::{eyre, Result};
+use color_eyre::Report;
+use log::warn;
+
+use crate::metadata::structs::Probe;
+
+const BMP_PRODUCT_STRING: &str = "Black Magic Probe";
+const BMP_PRODUCT_STRING_LENGTH: usize = BMP_PRODUCT_STRING.len();
+const NATIVE_BPM: &str = "native";
+
+pub enum ProbeIdentity
+{
+    ProbeVersion { name: String, version: String },
+    Probe {name: String},
+    None
+}
+
+fn probe_from_bpm(input: &str) -> Result<String>
```

Comment:
What does `bpm` stand for here? It's unclear and we can't figure out what you have in mind for it to expand to - please let's avoid using shortenings where possible and not already in the established project nomenclature.